generator/src/main/java/com/scottlogic/deg/generator/generation/fieldvaluesources/DateTime/RandomDateIterator.java

```diff
@@ -0,0 +1,35 @@
+package com.scottlogic.deg.generator.generation.fieldvaluesources.DateTime;
+
+import com.scottlogic.deg.generator.utils.RandomNumberGenerator;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Iterator;
+
+public class RandomDateIterator implements Iterator<LocalDateTime> {
+    private final LocalDateTime minDate;
+    private final LocalDateTime maxDate;
+    private final RandomNumberGenerator random;
+
+    RandomDateIterator(LocalDateTime minDate, LocalDateTime maxDate, RandomNumberGenerator randomNumberGenerator) {
+        this.minDate = minDate;
+        this.maxDate = maxDate;
+        this.random = randomNumberGenerator;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return true;
+    }
+
+    @Override
+    public LocalDateTime next() {
+        long min = this.minDate.toInstant(ZoneOffset.UTC).toEpochMilli();
+        long max = this.maxDate.toInstant(ZoneOffset.UTC).toEpochMilli() - 1;
+
+        long generatedLong = (long)random.nextDouble(min, max);
```

Comment:
Do we have a random dependency to make this mockable and easier to test?

Reply:
Yes there is a `RandomNumberGenerator` which can be mocked, see `TemporalFieldValueSourceTests` for an example